jib-core/src/main/java/com/google/cloud/tools/jib/http/Connection.java

```diff
@@ -87,41 +88,32 @@
    * Registers proxy credentials onto transport client, in order to deal with proxies that require
    * basic authentication.
    *
-   * @param transport
+   * @param transport Apache HTTP transport
    */
-  private static void addProxyCredentials(HttpTransport transport) {
-    DefaultHttpClient httpClient =
-        (DefaultHttpClient) ((ApacheHttpTransport) transport).getHttpClient();
-
-    boolean httpProxy = System.getProperty("http.proxyHost") != null;
-    boolean httpCreds =
-        System.getProperty("http.proxyUser") != null
-            && System.getProperty("http.proxyPassword") != null;
-    if (httpProxy && httpCreds) {
-      httpClient
-          .getCredentialsProvider()
-          .setCredentials(
-              new AuthScope(
-                  System.getProperty("http.proxyHost"),
-                  Integer.parseInt(System.getProperty("http.proxyPort", "8080"))),
-              new UsernamePasswordCredentials(
-                  System.getProperty("http.proxyUser"), System.getProperty("http.proxyPassword")));
-    }
-    boolean httpsProxy = System.getProperty("https.proxyHost") != null;
-    boolean httpsCreds =
-        System.getProperty("https.proxyUser") != null
-            && System.getProperty("https.proxyPassword") != null;
-    if (httpsProxy && httpsCreds) {
-      httpClient
-          .getCredentialsProvider()
-          .setCredentials(
-              new AuthScope(
-                  System.getProperty("https.proxyHost"),
-                  Integer.parseInt(System.getProperty("https.proxyPort", "443"))),
-              new UsernamePasswordCredentials(
-                  System.getProperty("https.proxyUser"),
-                  System.getProperty("https.proxyPassword")));
+  private static void addProxyCredentials(ApacheHttpTransport transport) {
+    addProxyCredentials(transport, "https");
+    addProxyCredentials(transport, "http");
+  }
+
+  private static void addProxyCredentials(ApacheHttpTransport transport, String protocol) {
+    Preconditions.checkArgument(protocol.equals("http") || protocol.equals("https"));
+
+    String proxyHost = System.getProperty(protocol + ".proxyHost");
+    String proxyUser = System.getProperty(protocol + ".proxyUser");
+    String proxyPassword = System.getProperty(protocol + ".proxyPassword");
+    if (proxyHost == null || proxyUser == null || proxyPassword == null) {
+      return;
     }
+
+    String defaultProxyPort = protocol.equals("http") ? "80" : "443";
+    int proxyPort = Integer.parseInt(System.getProperty(protocol + ".proxyPort", defaultProxyPort));
+
+    DefaultHttpClient httpClient = (DefaultHttpClient) transport.getHttpClient();
```

Comment:
Could we avoid this cast?

Reply:
Doesn't seem like we can. Need `DefaultHttpClient` for `getCredentialsProvider()`, which is not part of `HttpClient`. In our codebase, this is always a `DefaultHttpClient` instance.

Reply:
A little curious if switching to https://github.com/googleapis/google-http-java-client/tree/master/google-http-client-apache will give us an updated api for this?

Reply:
We are using Google HTTP Client. There might be an alternative way to set a proxy and credentials (or not really, don't know), but this is a proven way at least.

Reply:
Yeah, sorry I guess my comment wasn't clear. I tried finding a solution using the current version, but the ApacheHttpTransport (and Builder) objects don't have a good passthrough for setting proxy credentials, which is a shame.

I was hoping, give that they are doing work on this, that we might be able to use new features, but I'm not sure exactly what their plan is.

Reply:
Anyway, we can ignore my comments for now.